Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityTeleport.cs

```diff
@@ -27,16 +27,28 @@ namespace HoloToolkit.Unity.InputModule
     public class MixedRealityTeleport : Singleton<MixedRealityTeleport>, IControllerInputHandler
     {
         [Tooltip("Name of the thumbstick axis to check for teleport and strafe.")]
-        public string LeftThumbstickX = "CONTROLLER_LEFT_STICK_HORIZONTAL";
+        public XboxControllerMappingTypes HorizontalStrafe = XboxControllerMappingTypes.XboxLeftStickHorizontal;
 
         [Tooltip("Name of the thumbstick axis to check for teleport and strafe.")]
-        public string LeftThumbstickY = "CONTROLLER_LEFT_STICK_VERTICAL";
+        public XboxControllerMappingTypes VerticalStrafe = XboxControllerMappingTypes.XboxLeftStickVertical;
 
         [Tooltip("Name of the thumbstick axis to check for rotation.")]
-        public string RightThumbstickX = "CONTROLLER_RIGHT_STICK_HORIZONTAL";
+        public XboxControllerMappingTypes HorizontalRotation = XboxControllerMappingTypes.XboxRightStickHorizontal;
 
         [Tooltip("Name of the thumbstick axis to check for rotation.")]
-        public string RightThumbstickY = "CONTROLLER_RIGHT_STICK_VERTICAL";
+        public XboxControllerMappingTypes VerticalRotation = XboxControllerMappingTypes.XboxRightStickVertical;
+
+        [Tooltip("Custom Input Mapping for horizontal teleport and strafe")]
+        public string LeftThumbstickX = InputMappingAxisUtility.CONTROLLER_LEFT_STICK_HORIZONTAL;
```

Comment:
Do we need all these things seemingly mapped twice? HorizontalStrafe <-> LeftThumbstickX, HorizontalRotation <-> RightThumbstickX etc.

Reply:
It's for custom mappings.

Reply:
Also so there's not any breaking changes.

Reply:
Should `useCustomMapping` be true by default to prevent the breaking change? Otherwise it uses the new mappings by default, if I'm understanding correctly.

Reply:
No, it's not a breaking change because the string value isn't different, and the value serialized in the prefab isn't any different either.

Reply:
![image](https://user-images.githubusercontent.com/13334553/33691889-70aa40e6-da9f-11e7-8921-fa04c1ae8db1.png)
![image](https://user-images.githubusercontent.com/13334553/33691896-7ef348aa-da9f-11e7-8a52-5ab3f008c640.png)

Reply:
If I've changed the string axes to my own custom values already, this will break me.

Reply:
Shouldn't if your serialization is saved in the scene and not your prefab

Reply:
But the `useCustomMappings` flag being set to false by default will ignore that serialization and instead use the drop-downs by default.

Reply:
I've got an idea about how to fix this. One Sec.

Reply:
Checkout 8ed07d0

Reply:
That seems overly complicated vs setting `useCustomMappings` to true by default. Is there a reason we shouldn't do that?

If I've previously used the custom mappings, but want to switch to using the drop-downs, I'll have to reset my custom mappings exactly or this will switch me over every time I load the project.

Reply:
It's better for users to see the enum.  Simpler and more straight forward.

Reply:
Trying to keep the newb in mind

Reply:
>If I've previously used the custom mappings, but want to switch to using the drop-downs, I'll have to reset my custom mappings exactly or this will switch me over every time I load the project.

If you reset the prefab or remove the custom mapping completely and leave an empty field it will not do this.

Reply:
Yeah. Needing to reset the axes exactly is a minor inconvenience for those benefits

Reply:
I could also add a simple reset button that will fix that too

Reply:
> But it's essentially the same as just resetting the component.

Agreed. Could be a good addition later, but I don't consider that blocking. There are definitely other ways to reset.

Reply:
I thought about making it so when you switch to the drop downs it would reset the string values, but I thought that was too intrusive, in case you wanted to switch back.

Reply:
Agreed, if I'm just unchecking the box to see what it does.